Messaging.md

```diff
@@ -44,15 +55,102 @@ The following convenience types are also defined:
 * `contract_id`: a 32-byte contract_id (see [Protocol Specification](Protocol.md))
 * `sha256`: a 32-byte SHA2-256 hash
 * `signature`: a 64-byte bitcoin Elliptic Curve signature
+* `ecdsa_adaptor_signature`: a 162-byte ECDSA adaptor signature (TODO: link to doc once [#50](https://github.com/discreetlogcontracts/dlcspecs/issues/50) is done)
+* `x_point`: a 32-byte x-only public key with implicit y-coordinate being even as in [BIP 340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#design)
```

Comment:
More for my understanding, but does this need to be wit implicit y-coordinate? I wanted to avoid exposing new types in my implementation so I just ended up using regular public key (33-bytes), so wondering if there could be an issue.

Reply:
AFIAK you should be able to just drop the first byte of the pubkey.

Reply:
Yeah that's what I do currently :) I guess it must be fine

Reply:
yeah ... the translation between x-only and normal is adding or subtracting a `02` byte on the front, but I figured it would be nice not to include the `02` on Schnorr keys ... open to changing this though

Reply:
I feel it depends if x-only pubkeys are going to be something common in the (near) future or if will still be mainly internal to libsecp (or other crypto libraries).

Reply:
I think either way if someone wants to support DLC spec stuff they are required to implement Schnorr interfaces which all use x-only so I don't think its a weird ask

Reply:
I personally implemented the Schnorr module in rust using just regular public keys, because I didn't want to add extra types to the public interface and to keep things simple(r). If I add x-only pubkeys to the public interface, I also need to start adding util functions to get an x-only pubkey from private key, or from a regular public key... It's not infeasible, but it didn't seem super necessary either. And since they provide a `xonly_pubkey_from_pubkey` function in libsecp I thought it was maybe not meant to be exposed externally.

Reply:
tbh I don't know what the internal/external plan is for x-only but I know that on-chain there will be x-only keys and nonces once we have taproot so I think the types will be pretty prevalent in libraries. Furthermore I'd argue that since this is only the P2P part of the protocol you can just use whatever type you want internally and only do x-only conversions in your serialization/deserialization logic as it is pretty trivial.

Reply:
Ok let's go with that for now then. I was also trying to understand if I should switch to x-only in my libraries but I guess I can figure that out later.